homeassistant/components/zha/switch.py

```diff
@@ -138,3 +150,118 @@ async def async_update(self) -> None:
 
         self._state = len(on_states) > 0
         self._available = any(state.state != STATE_UNAVAILABLE for state in states)
+
+
+class ZHASwitchConfigurationEntity(ZhaEntity, SwitchEntity):
+    """Representation of a ZHA switch configuration entity."""
+
+    _zcl_attribute: str
+    _zcl_inverter_attribute: str = ""
+
+    @classmethod
+    def create_entity(
+        cls,
+        unique_id: str,
+        zha_device: ZHADevice,
+        channels: list[ZigbeeChannel],
+        **kwargs,
+    ) -> ZhaEntity | None:
+        """Entity Factory.
+
+        Return entity if it is a supported configuration, otherwise return None
+        """
+        channel = channels[0]
+        if (
+            cls._zcl_attribute in channel.cluster.unsupported_attributes
+            or channel.cluster.get(cls._zcl_attribute) is None
+        ):
+            _LOGGER.debug(
+                "%s is not supported - skipping %s entity creation",
+                cls._zcl_attribute,
+                cls.__name__,
+            )
+            return None
+
+        return cls(unique_id, zha_device, channels, **kwargs)
+
+    def __init__(
+        self,
+        unique_id: str,
+        zha_device: ZHADevice,
+        channels: list[ZigbeeChannel],
+        **kwargs,
+    ) -> None:
+        """Init this number configuration entity."""
+        self._channel: ZigbeeChannel = channels[0]
+        super().__init__(unique_id, zha_device, channels, **kwargs)
+
+    async def async_added_to_hass(self) -> None:
+        """Run when about to be added to hass."""
+        await super().async_added_to_hass()
+        self.async_accept_signal(
+            self._channel, SIGNAL_ATTR_UPDATED, self.async_set_state
+        )
+
+    @callback
+    def async_set_state(self, attr_id: int, attr_name: str, value: Any):
+        """Handle state update from channel."""
+        self.async_write_ha_state()
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the switch is on based on the statemachine."""
+        val = bool(self._channel.cluster.get(self._zcl_attribute))
+        invert = bool(self._channel.cluster.get(self._zcl_inverter_attribute))
+        return (not val) if invert else val
+
+    async def async_turn_on_off(self, state) -> None:
```

Comment:
Missing type annotation for the `state` parameter. Please type the whole signature when adding type annotations.

```py
async def async_turn_on_off(self, state: bool) -> None:
```

Reply:
Sorry I missed this when I reviewed it.

Reply:
No, worries. 👍

Reply:
added the type info to another PR I have open in this commit: https://github.com/home-assistant/core/pull/71527/commits/86dfa12442005e852c76cc5e4bda741ad044f78f